app/src/main/java/com/cp/tasky/auth/login/presentation/LoginScreen.kt

```diff
@@ -60,7 +64,9 @@ private fun LoginScreen(
     modifier: Modifier = Modifier,
     viewState: LoginScreenState,
     onEvents: (LoginScreenEvent) -> Unit,
-    networkState: Result<LoginResponse, Error>
+    networkState: Result<LoginResponse, Error>,
```

Comment:
I'd argue that the network state isn't really something the UI should deal with. Instead I would keep it as dumb as possible with as little parsing/processing logic as possible. This logic can go in the ViewModel instead which can map the network state result to a corresponding UI state property (like `isLoggedIn` or `errorMessage` or so)

The reason is simply that it's a lot easier to test ViewModel code than UI code. The latter requires you to boot up an Android device, simulate a UI environment and then finally test your logic. Testing a ViewModel all works on the JVM without needing an Android device at all.

Reply:
addressed